### PR TITLE
feat(z): add z cache results to completion

### PR DESCRIPTION
**Pull request by sghsri**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**

The current behavior is that z only shows the paths in the current folder. 

**What is the new behavior (if this is a feature change)?**

The PR changes the z completion to take into account the directories cached by z and the weighting that z provides them. Now you will be able to see all the possible paths that z can take you.

image

**Additional info:**

### CODE REVIEW

File: src/z.ts [hunk 1]
```diff
@@ -3,10 +3,53 @@ const completionSpec: Fig.Spec = {
...
+          // that's already been entered. This 'if' prevents redundant
+          // suggestions.
+          if (!args.includes(name)) {
+            return {
+              name,
+              description: path,
+              // Docs state max weight is 100 but this seems
+              // to work regardless of any amount over that limit.
+              // Fig should defer assigning priority to z.
+              // 75 added to keep args above options.
```
[the remark this answers is not in the document]
Comment (sghsri) [reply src/z.ts:46]: ok, good call! just pushed up a commit doing just that

File: src/z.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ const completionSpec: Fig.Spec = {
               // to work regardless of any amount over that limit.
               // Fig should defer assigning priority to z.
               // 75 added to keep args above options.
-              priority: 75 + weight,
+              priority: 75 + (weight >= 100 ? 100 : weight),
```
Comment (fedeci) [inline src/z.ts:47]: This isn't really what I was thinking. My idea was more like this: the max sum of all the frecencies of z is either a custom value or 9000 https://github.com/rupa/z/blob/b82ac78a2d4457d2ca09973332638f123f065fd1/z.sh#L72
We should assume that all the items are going to be in the range `[0, MAX || 9000]`.
This should do the tweak:
```suggestion
              priority: 75 + (weight * 25 / (maxFrecency || 9000)), // where maxFrecency is read executing a shell command: `echo "$_Z_MAX_SCORE"`
```
Ping me for help!

Comment (sghsri) [reply src/z.ts:47]: oooh that makes sense! how would we go about executing another shell command here?

Comment (sghsri) [reply src/z.ts:47]: also, when echoing that shell variable even though I have z active, its empty 
.

Comment (sghsri) [reply src/z.ts:47]: I'm using https://github.com/agkozak/zsh-z , which is said to be more stable and more performant

Comment (fedeci) [reply src/z.ts:47]: Updated!

Comment (sghsri) [reply src/z.ts]: Ok awesome, pushed up the commit with the suggestion!

**Review by fedeci — APPROVED** [review]
Thanks again for contributing!

**Comment by fedeci** [comment]
@sghsri linter is failing :( Can you run `npm run lint:fix` and commit the changes?

**Comment by sghsri** [comment]
> @sghsri linter is failing :( Can you run `npm run lint:fix` and commit the changes?
commit after running `npm run lint:fix` pushed up!

**Comment by sghsri** [comment]
@fedeci thanks for approving! What is the next step in the process for getting this merged in / available in the production fig build?

**Comment by fedeci** [comment]
Merged!

**Pull request merged**